### ISSUE TITLE
test_logging: test_post_fork_child_no_deadlock() failed with timeout on AMD64 Arch Linux Asan Debug 3.10

### ISSUE THREAD

**Issue by vstinner** [python/cpython#90867]
BPO | [46711](https://bugs.python.org/issue46711)
--- | :---
Nosy | @vstinner, @sobolevn
PRs | <li>python/cpython#31274</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-02-10.13:43:47.841>
labels = ['tests', '3.10']
title = 'test_logging: test_post_fork_child_no_deadlock() failed with timeout on AMD64 Arch Linux Asan Debug 3.10'
updated_at = <Date 2022-02-11.09:48:27.341>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2022-02-11.09:48:27.341>
actor = 'sobolevn'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Tests']
creation = <Date 2022-02-10.13:43:47.841>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 46711
keywords = ['patch']
message_count = 1.0
messages = ['413000']
nosy_count = 2.0
nosy_names = ['vstinner', 'sobolevn']
pr_nums = ['31274']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue46711'
versions = ['Python 3.10']
```

</p></details>

* PR: gh-99071

* PR: gh-99098

* PR: gh-99099

**Comment by hauntsaninja** [comment]
Thanks, looks like we've completed this

**Comment by vstinner** [comment]
Yep, I fixed it with commit f09da28768b77713566e932e912f107b6b57e8fd, backported to 3.11 and 3.10.
